cmd/admin-handlers.go

```diff
@@ -801,6 +810,91 @@ func convertValueType(elem []byte, jsonType gjson.Type) (interface{}, error) {
 	}
 }
 
+func isInvalidJSONField(expectedKeys gjson.Result, elem []byte) (string, string) {
+	keys := string(elem)
+	var found bool
+	var fullKeyName string
+	// Validation for a single key
+	if expectedKeys.Type == gjson.Null &&
+		expectedKeys.Raw == "" &&
+		expectedKeys.Str == "" &&
+		expectedKeys.Num == 0 &&
+		expectedKeys.Index == 0 {
+		return "Invalid key", ""
+	}
+	// Validation for multiple key names in JSON format
+	var skipChrsFront int
+	var skipChrsEnd int
+	if expectedKeys.Type == gjson.JSON {
+		if len(keys) == 2 {
+			return "Invalid/Empty JSON data", ""
+		}
+		raw := expectedKeys.Raw
+		if keys[:1] == "{" {
+			skipChrsFront = 1
+		} else {
+			skipChrsFront = 0
+		}
+		if keys[len(keys)-1:] == "}" {
+			skipChrsEnd = 1
+		} else {
+			skipChrsEnd = 0
+		}
+		r := regexp.MustCompile(`[^\s":]+|"([^"]*)"`)
+		keys := r.FindAllString(keys[skipChrsFront:len(keys)-skipChrsEnd], -1)
+		rawKeys := regexp.MustCompile(`[,:]`).Split(raw[1:len(raw)-1], -1)
+
+		for i := 0; i < len(keys); i++ {
+			if keys[i][:1] == "{" {
+				if keys[i][1:2] == "\"" {
+					skipChrsFront = 2
+				} else {
+					if keys[i][len(keys[i])-1:] == "\"" {
+						skipChrsEnd = 1
+					} else {
+						skipChrsEnd = 0
+					}
+					return "Invalid JSON format", fullKeyName + "." + keys[i][1:len(keys[i])-skipChrsEnd]
```

Comment:
This shouldn't be a string instead should be errors.New() or `fmt.Errorf`

Reply:
OK